ivre/web/utils.py

```diff
@@ -138,7 +137,7 @@ def query_from_params(params):
             else [neg, pval, None]
             for neg, pval in (
                 (True, x[1:]) if x[:1] in '!-' else (False, x)
-                for x in shlex.split(query)
+                for x in query.split()
```

Comment:
Is this related to this PR?

Reply:
Yes! `shlex` causes strange behaviors in web filters. For example, if one wants to match the regex `\d*`, one would use `<something>:/\d*/`. `shlex.split()` interprets the given string as a "shell-like" command and transforms `\d` in `d`. Thus, the pattern checked is `/d*/` which has a significantly different meaning. Currently, the workaround is to escape `\` like `/\\d*/`, which does not seem user friendly to me.

Reply:
I get it. The problem here is that `shlex.split()` is quite different than `str.split()`:
```
In [1]: "m1 'm2 m3'".split()                                                                                                                                   
Out[1]: ['m1', "'m2", "m3'"]

In [2]: import shlex                                                                                                                                           

In [3]: shlex.split("m1 'm2 m3'")                                                                                                                              
Out[3]: ['m1', 'm2 m3']
```

Reply:
You're right! Perhaps the right solution would be to escape all `\` automatically before calling `shlex.split` : 
```python
     try:
        query = query.replace('\\', '\\\\')
        return [
            [neg] + pval.split(':', 1) if ':' in pval
            else [neg, pval, None]
            for neg, pval in (
                (True, x[1:]) if x[:1] in '!-' else (False, x)
                for x in shlex.split(query)
            )
        ]
```
Does it feel right to you?

Reply:
Could be. We have to think about this and check if we are not forgetting other corner cases.

Reply:
I'm going to update the code. We'll see if one of the tests fails (every corner cases are not tested, but still it's a first step).